### PR TITLE
enhance: optimize milvus core building

### PR DESCRIPTION
**Pull request by chyezh**
issue: #35549,#35611,#35633
pr: #35610

- remove milvus_segcore milvus_indexbuilder..., add libmilvus_core
- core building only link once
- move opendal compilation into cmake
- fix odr

### CODE REVIEW

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

File: internal/core/src/log/Log.cpp [hunk 1]
```diff
@@ -1,131 +0,0 @@
-// Licensed to the LF AI & Data foundation under one
```
Comment (chyezh) [inline internal/core/src/log/Log.cpp:1]: repeated object file with milvus-storage

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**